ZKProof/Plonkish/Defs.lean

```diff
@@ -0,0 +1,237 @@
+import Mathlib.Data.Rel
+import Mathlib.Data.Fin.Basic
+import Mathlib.Logic.Basic
+import Mathlib.Logic.Function.Defs
+import Mathlib.Algebra.Group.Defs
+import Mathlib.Algebra.Field.Defs
+import Mathlib.Tactic
+
+import ZKProof.Relations
+
+variable (F : Type) [Field F]
+
+
+/-- The location of a witness entry. -/
+structure Location (m n : ℕ) where
+  /-- Column number. -/
+  i : Fin m
+  /-- Row number. -/
+  j : Fin n
+deriving DecidableEq
+
+/-- Utility function to construct a `Location`. -/
+def entry {m n : ℕ} (i : Fin m) (j : Fin n) : Location m n := { i := i, j := j }
+
+/-- An abstract Plonkish circuit. -/
+structure AbstractCircuit where
+  /-- Length of the instance vector. -/
+  t : ℕ
+
+  /-- Number of columns for the witness matrix. -/
+  m : ℕ
```

Comment:
Could we replace "m:ℕ" with "m:ℕ+" and remove the following line?

Reply:
I was mistaken on the call; actually this works fine. (A couple of other changes are needed for the types of `m` and `n` to be consistent everywhere.)